bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -373,14 +379,30 @@ WebViewEnvironment createEnvironment() {
 
 @Override
 public void create(Composite parent, int style) {
-	checkDeadlock();
 	containingEnvironment = createEnvironment();
 
 	long[] ppv = new long[1];
 	int hr = containingEnvironment.environment().QueryInterface(COM.IID_ICoreWebView2Environment2, ppv);
 	if (hr == COM.S_OK) environment2 = new ICoreWebView2Environment2(ppv[0]);
 
+	// If leads to deadlock then execute asynchronously and move on.
+	// The webview calls are queued to be executed when it is done executing the current task.
+	if (leadsToDeadlock() && !asyncExecution) {
+		asyncExecution = true;
+		browser.getDisplay().asyncExec(() -> setupBrowser());
```

Comment:
A consequence of this change is that after calling `create()` there is no guarantee anymore that the browser has been completely initialized. So subsequent calls to other methods of the instance might fail. E.g., imagine `setupBrowser()` is performed asynchronously and someone calls `execute()` on the instance, then the call will fail as `webView` might still be `null`.
Wouldn't this asynchronous initialization require all methods to be guarded with a check that initialization has been completed?

Reply:
To my knowledge there are only setUrl, setText and execute which need the broswer environment (the things initialized by setupBrowser). I have guarded the 2 but execute is still a subject of discussion because of its synchronous behaviour. So I haven't done any changes there yet. But, since we are going to deal with execute in a follow up issue, I can put a temporary guard to fail execute on (if called during a deadlock), this is one solution I have in my mind for now.